src/adapters/ui/iconview.js

```diff
@@ -127,6 +138,7 @@ const createShortcuts = (root, readfile, writefile) => {
   };
 
   const add = entry => read(root)
+    .then(updateShortcutLabel(core))
```

Comment:
Since there was no unit tests on this module (sadly), I went ahead and tested this implementation.

And this caused an exception (before review as well).

Should probably be:

```
    .then(shortcuts => shortcuts.map(updateShortcutLabel(core)))
```

Reply:
Oh, so it does!